// Code generated by mockery v2.46.3. DO NOT EDIT.

package mocks

import mock "github.com/stretchr/testify/mock"

// Retry is an autogenerated mock type for the Retry type
type Retry struct {
	mock.Mock
}

type Retry_Expecter struct {
	mock *mock.Mock
}

func (_m *Retry) EXPECT() *Retry_Expecter {
	return &Retry_Expecter{mock: &_m.Mock}
}

// Execute provides a mock function with given fields: wrapped
func (_m *Retry) Execute(wrapped func() error) error {
	ret := _m.Called(wrapped)

	if len(ret) == 0 {
		panic("no return value specified for Execute")
	}

	var r0 error
	if rf, ok := ret.Get(0).(func(func() error) error); ok {
		r0 = rf(wrapped)
	} else {
		r0 = ret.Error(0)
	}

	return r0
}

// Retry_Execute_Call is a *mock.Call that shadows Run/Return methods with type explicit version for method 'Execute'
type Retry_Execute_Call struct {
	*mock.Call
}

// Execute is a helper method to define mock.On call
//   - wrapped func() error
func (_e *Retry_Expecter) Execute(wrapped interface{}) *Retry_Execute_Call {
	return &Retry_Execute_Call{Call: _e.mock.On("Execute", wrapped)}
}

func (_c *Retry_Execute_Call) Run(run func(wrapped func() error)) *Retry_Execute_Call {
	_c.Call.Run(func(args mock.Arguments) {
		run(args[0].(func() error))
	})
	return _c
}

func (_c *Retry_Execute_Call) Return(_a0 error) *Retry_Execute_Call {
	_c.Call.Return(_a0)
	return _c
}

func (_c *Retry_Execute_Call) RunAndReturn(run func(func() error) error) *Retry_Execute_Call {
	_c.Call.Return(run)
	return _c
}

// NewRetry creates a new instance of Retry. It also registers a testing interface on the mock and a cleanup function to assert the mocks expectations.
// The first argument is typically a *testing.T value.
func NewRetry(t interface {
	mock.TestingT
	Cleanup(func())
}) *Retry {
	mock := &Retry{}
	mock.Mock.Test(t)

	t.Cleanup(func() { mock.AssertExpectations(t) })

	return mock
}
